src/xdist/scheduler/protocol.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+from typing import Protocol
+from typing import Sequence
+
+from xdist.workermanage import WorkerController
+
+
+class Scheduling(Protocol):
```

Comment:
Minor, up to you: I think a more appropriate name would be `Scheduler` (a subject), "sheduling" is a verb AFAIK. But I understand also that all the implementations use `Scheduling` in their name, so on one hand it makes sense to name this `Scheduling`, on the other hand, "two wrongs don't make a right".

Up to you really. 👍

Reply:
Yes, I don't know why the existing schedulers are called `Scheduling` but I don't think we can safely change them at this point. And then I think being consistent is better.